packages/pn-commons/src/components/AppStatus/downtimeLog.utils.tsx

```diff
@@ -0,0 +1,136 @@
+import { useCallback } from 'react';
+import { Button, Chip, Stack, Typography, useTheme } from '@mui/material';
+import DownloadIcon from '@mui/icons-material/Download';
+import { CardElement, Column, Item } from '../../types';
+import { DowntimeLogPage, DowntimeStatus } from '../../models';
+import { getLocalizedOrDefaultLabel } from '../../services/localization.service';
+import { formatDateTime } from '../../services/date.service';
+
+export function booleanStringToBoolean(booleanString: string): boolean {
+  return booleanString.toLowerCase() === "true";
+}
+
+/* eslint-disable-next-line arrow-body-style */
+const FormattedDateAndTime = ({ date, inTwoLines }: { date: string; inTwoLines?: boolean }) => {
+  if (date) {
+    const dateAndTime = formatDateTime(date);
+    return inTwoLines
+      ? <Stack direction="column">
+        <Typography variant="body2">{dateAndTime.date},</Typography>
+        <Typography variant="body2">{dateAndTime.time}</Typography>
+      </Stack>
+      : <Typography variant="body2">{dateAndTime.date}, {dateAndTime.time}</Typography>;
+  } else {
+    return <Typography variant="body2">-</Typography>;
+  }
+};
+
+export function adaptFieldSpecToMobile(desktopFieldSpec: Omit<Column<DowntimeLogColumn>, "width">): CardElement {
+  return {
+    id: desktopFieldSpec.id,
+    label: desktopFieldSpec.label,
+    getLabel: desktopFieldSpec.getCellLabel,
+  };
+}
+
+
+export type DowntimeLogColumn =
+  | 'startDate'
+  | 'endDate'
+  | 'functionality'
+  | 'legalFactId'
+  | 'status'
+  | '';
+
+
+export function useFieldSpecs({ getDowntimeLegalFactDocumentDetails }: { getDowntimeLegalFactDocumentDetails: (legalFactId: string) => any}) {
+  const theme = useTheme();
+
+  const getDateFieldSpec = useCallback((fieldId: DowntimeLogColumn, inTwoLines: boolean): Omit<Column<DowntimeLogColumn>, "width"> => ({
```

Comment:
Because we return object, I would replace useCallback (for functions) with useMemo (for variables).
This for all useCallback below

Reply:
The four `get...FieldSpec`, as well as `getRows`, are indeed functions. Please check how they are used in `DesktopDowntimeLog.tsx` and `MobileDowntimeLog.tsx`. Therefore, I still guess that `useCallback` is the right hook for these definitions.